server/src/test/java/org/elasticsearch/common/settings/SettingTests.java

```diff
@@ -53,34 +54,65 @@ public void testGet() {
     }
 
     public void testByteSize() {
-        Setting<ByteSizeValue> byteSizeValueSetting =
-            Setting.byteSizeSetting("a.byte.size", new ByteSizeValue(1024), Property.Dynamic, Property.NodeScope);
-        assertFalse(byteSizeValueSetting.isGroupSetting());
-        ByteSizeValue byteSizeValue = byteSizeValueSetting.get(Settings.EMPTY);
-        assertEquals(byteSizeValue.getBytes(), 1024);
+        {
+            final Setting<ByteSizeValue> byteSizeValueSetting =
+                    Setting.byteSizeSetting("a.byte.size", new ByteSizeValue(1024), Property.Dynamic, Property.NodeScope);
+            assertFalse(byteSizeValueSetting.isGroupSetting());
+            final ByteSizeValue byteSizeValue = byteSizeValueSetting.get(Settings.EMPTY);
+            assertThat(byteSizeValue.getBytes(), equalTo(1024L));
+        }
 
-        byteSizeValueSetting = Setting.byteSizeSetting("a.byte.size", s -> "2048b", Property.Dynamic, Property.NodeScope);
-        byteSizeValue = byteSizeValueSetting.get(Settings.EMPTY);
-        assertEquals(byteSizeValue.getBytes(), 2048);
+        // byte size value setting is less than minimum value for setting
+        {
```

Comment:
Personally I would rather do these as separate test methods (eg testByteSizeMin here), instead of blocks.